src/arrayMethodJoin.js

```diff
@@ -4,9 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const validElementsList = [];
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+
+      if (element !== null && element !== undefined) {
+        validElementsList.push(element);
+      } else {
+        validElementsList.push('');
+      }
+    }
+
+    for (let i = 0; i < validElementsList.length; i++) {
+      result += validElementsList[i];
+
+      if (i < validElementsList.length - 1) {
+        result += separator;
+      }
+    }
```

Comment:
validElementsList - is redudant.
No need in 2 loops
```suggestion

    for (let i = 0; i < this.length; i++) {
      const element = this[i];

      if (element !== null && element !== undefined) {
        result += element ;
      }
      
      if (i < this.length - 1) {
        result += separator;
      }
    }
```